api/api/src/main/scala/uk/ac/wellcome/platform/api/QueryParams.scala

```diff
@@ -219,6 +220,14 @@ trait QueryParamsUtils extends Directives with TimeInstances {
         .getOrElse(Left(invalidValuesMsg(List(str), values.map(_._1).toList)))
     }
 
+  def decodeOneWithDefaultOf[T](default: T, values: (String, T)*): Decoder[T] =
```

Comment:
I think this could be written more succinctly as:
```
Decoder.decodeString.map { values.toMap.getOrElse(_, default) }
```
since we are always giving `emap` a `Right`

Reply:
Maybe the function signature should be like this, so you can explicitly request the default value also:

```scala
def decodeOneWithDefaultOf[T](default: (String, T), otherValues: (String, T)*): Decoder[T]
```

Reply:
@jamieparkinson 👍 

@warrd we don't actually have a string for requesting the default. None or wrong = default in this case.

Reply:
I think that might confuse me as a user of an API, where you cannot specify the default explicitly. I know this is not a public API, but maybe worth reconsidering in the case that this functionality ends up being used elsewhere.